holoviews/tests/plotting/bokeh/test_links.py

```diff
@@ -53,8 +50,41 @@ def test_range_tool_link_callback_single_axis_overlay_target_image_source(self):
         self.assertEqual(range_tool.x_range, tgt_plot.handles['x_range'])
         self.assertIs(range_tool.y_range, None)
 
+    def test_range_tool_link_callback_single_axis_curve_target_image_dmap_source(self):
+        # Choosing Image to exert the apply_nodata compositor
+        src = DynamicMap(
+            lambda a: Image(a*np.random.random((20, 20)), bounds=[0, 0, 9, 9]),
+            kdims=['a']
+        ).redim.range(a=(0.1,1))
+        target = Curve(np.arange(10))
+        RangeToolLink(src, target)
+        layout = target + src
+        plot = bokeh_renderer.get_plot(layout)
+        tgt_plot = plot.subplots[(0, 0)].subplots['main']
+        src_plot = plot.subplots[(0, 1)].subplots['main']
+        range_tool = src_plot.state.select_one({'type': RangeTool})
+        self.assertEqual(range_tool.x_range, tgt_plot.handles['x_range'])
```

Comment:
I'm slowly trying to move away from unittest to pytest (will take time). So I would prefer to use `assert` here, even though you are mimicking the existing code.

Reply:
Done in https://github.com/holoviz/holoviews/pull/6003/commits/c6460fc03cbacec592d20c0f234fbaa63421e5ec